Tasks/AzureRmWebAppDeploymentV4/Tests/AzureRmWebAppDeploymentProviderL0Tests.ts

```diff
@@ -0,0 +1,58 @@
+import tl = require('vsts-task-lib');
+import { DeploymentFactory } from '../deploymentProvider/DeploymentFactory';
+import { AzureRmWebAppDeploymentProvider } from '../deploymentProvider/AzureRmWebAppDeploymentProvider'
+import { IWebAppDeploymentProvider } from '../deploymentProvider/IWebAppDeploymentProvider';
+import { TaskParametersUtility, TaskParameters, DeploymentType } from '../operations/TaskParameters';
+import { stringify } from 'querystring';
+import { PackageType } from 'webdeployment-common/packageUtility';
+import { getMockEndpoint } from '../node_modules/azure-arm-rest/Tests/mock_utils';
+import { mockAzureARMPreDeploymentSteps }  from "./mock_utils";
+
+getMockEndpoint();
+mockAzureARMPreDeploymentSteps();
+
+export class AzureRmWebAppDeploymentProviderL0Tests  {
+
+    public static async startAzureRmWebAppDeploymentProviderL0Tests() {
+        await AzureRmWebAppDeploymentProviderL0Tests.testForPreDeploymentSteps();
```

Comment:
Should we add tests for virtual application and post deployment step

Reply:
This is base class and for each subsequent extended class post deployment step is a part of DeployWebAppStep. Already added corresponding L0 for each subsequent extended class.  For virtual application I am adding L0.